scripts/fetch-icon.js

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env node
+
+const inquirer = require('inquirer');
+const xRay = require('x-ray')();
+const fetch = require('node-fetch');
+const { mkdirSync, writeFileSync, existsSync } = require('fs');
+const { join, extname } = require('path');
+const url = require('url');
+const makeDriver = require('request-x-ray');
+
+// mobile user agent because some sites only serve manifest/pwa for mobile
+const headers = {
+  'User-Agent':
+    'Mozilla/5.0 (Linux; Android 8.0; Pixel 2 Build/OPD3.170816.012) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Mobile Safari/537.36'
+};
+
+xRay.driver(makeDriver({ headers }));
+
+/**
+ *
+ * @param {string} size the size of the image, like "144x144"
+ * @returns the first dimension, like 144
+ */
+const parseSize = size => parseInt(size.split('x')[0]);
+
+/**
+ *
+ * @param {{sizes: string, src: string}[]} icons
+ */
+const findLargestIcon = icons => {
+  const icon = icons.reduce((largestIcon, icon) => {
+    if (
+      parseSize(icon.sizes) >
+      (largestIcon.sizes ? parseSize(largestIcon.sizes) : 0)
+    ) {
+      return icon;
+    }
+    return largestIcon;
+  });
+  return { icon: icon.src, size: icon.sizes };
+};
+
+/**
+ * @param {string} u the url to retrieve the name from
+ * @returns {string|undefined}
+ */
+const simplifyUrl = u => {
+  const { hostname } = url.parse(u);
+  if (hostname === null) {
+    return undefined;
+  }
+  return hostname
+    .replace(/\.[^.]*$/, '') // remove tld (last dot and everything after it)
+    .replace(/^.*\./, '') // remove subdomains (everything up to the last dot)
+    .toLowerCase();
+};
+
+module.exports = simplifyUrl;
+
+const main = async () => {
+  const { appUrl } = await inquirer.prompt([
+    {
+      name: 'appUrl',
+      type: 'input',
+      message: 'Link to a page in the PWA'
```

Comment:
Should we give a quick example? I tried `cloudfour.com` and it failed for me. Maybe something like this?
```
message: 'PWA URL (e.g. https://www.cloudfour.com):'
```
Is that too much? 😬